web/src/avatar.js

```diff
@@ -34,11 +35,32 @@ export function build_bot_create_widget() {
     );
 }
 
-export function build_bot_edit_widget($target) {
+export function build_bot_edit_widget($target, bot_details) {
     const get_file_input = function () {
         return $target.find(".edit_bot_avatar_file_input");
     };
 
+    if (bot_details.avatar_source === "G") {
+        $("#edit-bot-avatar-upload-widget .image-hover-background").hide();
+        $("#edit-bot-avatar-upload-widget .image-delete-button").hide();
+    }
+
+    $("#edit-bot-avatar-upload-widget .image-delete-button").on("click", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        $("#edit-bot-avatar-upload-widget .upload-spinner-background").css({visibility: "visible"});
+        $("#edit-bot-avatar-upload-widget .image-hover-background").hide();
+        $("#edit-bot-avatar-upload-widget .image-delete-button").hide();
+
+        bot_details.avatar_source = "G";
```

Comment:
Instead of doing, we can probably have a hidden input with stores value of avatar source. Mutating the object passed to function is not the best solution I think.

Reply:
fixed this in the updated PR.